capstone-backend/src/main/java/com/google/sps/data/BookList.java

```diff
@@ -17,77 +17,89 @@
  */
 public final class BookList {
 
-  private Long id;
-  private Long userID;
-  private String bookshelfID;
-  private Collection<Long> collaboratorsIDs;
+  private String id;
+  private String userID;
+  private Collection<String> gbookIDs;
+  private Collection<String> collaboratorsIDs;
 
-  public BookList(Long userID, String bookshelfID, Collection<Long> collaboratorsIDs, Long id) {
+  public BookList(String id, String userID, Collection<String> gbookIDs, Collection<String> collaboratorsIDs) {
+    this.id = id;
     this.userID = userID;
-    this.bookshelfID = bookshelfID;
+    this.gbookIDs = gbookIDs;
     this.collaboratorsIDs = collaboratorsIDs;
-    this.id = id;
   }
 
-  public BookList(Long userID, String bookshelfID, Collection<Long> collaboratorsIDs) {
-    this(userID, bookshelfID, collaboratorsIDs, null);
+  public BookList(String userID, Collection<String> gbookIDs, Collection<String> collaboratorsIDs) {
+    this("", userID, gbookIDs, collaboratorsIDs);
   }
 
-  public BookList(Long userID, String bookshelfID){
-    this(userID, bookshelfID, new HashSet<Long>(), null);
+  public BookList(String userID, Collection<String> gbookIDs) {
+    this("", userID, gbookIDs, new HashSet<String>());
   }
 
-  public void setBookshelf(String bookshelfID) {
-    this.bookshelfID = bookshelfID;
+  public void setGbookIDs(Collection<String> gbookIDs) {
+    this.gbookIDs = gbookIDs;
   }
 
-  public void setUserID(Long userID) {
+  public void setUserID(String userID) {
```

Comment:
Not sure if this should be final. Depends on if we want to let a user hand over ownership of a BookList.

Reply:
For right now, I will leave it final. I think we should limit the amount of variables we can change.